src/coreclr/nativeaot/Runtime/eventtrace.cpp

```diff
@@ -302,8 +325,47 @@ void EtwCallback(
         GCHeapUtilities::GetGCHeap()->DiagTraceGCSegments();
     }
 }
+
 #endif // FEATURE_ETW
 
+/**************************************************************************************/
+/* Called when ETW is turned OFF on an existing process .Will be used by the controller for end rundown*/
+/**************************************************************************************/
+void ETW::EnumerationLog::EndRundown()
```

Comment:
Should this also fire `DCEndInit` and `DCEndComplete`? I assume they are not needed for `dotnet-gcdump collect`, but they are part of the contract for how rundown works.

Reply:
They are basically indicators for tools to know when the rundown actually starts/completes (and the tool should have all the data, so can stop collecting). As long as they are fired before/after the rest of the end rundown events, it should be good.